vllm/core/scheduler.py

```diff
@@ -763,6 +763,84 @@ def _get_prompt_limit(self, seq_group: SequenceGroup) -> int:
         else:
             return prompt_limit
 
+    def _get_priority(self,
+                      seq_group: SequenceGroup) -> Tuple[Optional[int], float]:
+        """ Get the priority of the sequence group.
+        Highest preference to user-defined priority, followed by arrival time.
+        Args:
+            seq_group: The sequence group input.
+        Returns:
+            The priority of the sequence group.
+        """
+        return (seq_group.priority, seq_group.arrival_time)
+
+    def _schedule_priority_preemption(
+        self,
+        budget: SchedulingBudget,
+    ) -> int:
+        """Sorts waiting and running queue. Also, force preempt requests
+        from the running queue if their priority is lower.
+        Priority-based preemption is used with the priority policy.
+        Args:
+            budget: The scheduling budget. The argument is in-place updated
+                when any requests are scheduled.
+        Returns:
+            A count of priority-based preemptions.
+        """
+
+        running_queue = self.running
```

Comment:
nit: no need for intermediate variable here?

Reply:
Dequeue does not support in-place sort, so creating a new variable is required

Reply:
I just mean, you could remove this line and change the one below to
```python
running_queue = deque(sorted(self.running, key=self._get_priority))
```